service/history/historybuilder/event_factory.go

```diff
@@ -396,14 +396,20 @@ func (b *EventFactory) CreateWorkflowExecutionTerminatedEvent(
 func (b *EventFactory) CreateWorkflowExecutionOptionsUpdatedEvent(
 	versioningOverride *workflowpb.VersioningOverride,
 	unsetVersioningOverride bool,
+	attachRequestID string,
+	attachCompletionCallbacks []*commonpb.Callback,
```

Comment:
Still think you should just accept `*historypb.WorkflowExecutionOptionsUpdatedEventAttributes` here and avoid this function signature growing every time we add another attribute.

Reply:
It's a bit complicated. This is the event factory, ie., that's the function that builds `*historypb.WorkflowExecutionOptionsUpdatedEventAttributes`. I want to change to get the history request, but then it would add some logic to translate the request input to event attributes, which is not something we did so far.